lib/index.js

```diff
@@ -16,204 +16,175 @@ internals.defaults = {
+        return this.isConnected;
     }
 
-    if (name.indexOf('\0') !== -1) {
-        return new Error('Includes null character');
-    }
+    validateSegmentName(name) {
 
-    if (name.indexOf('system.') === 0) {
-        return new Error('Begins with "system."');
-    }
+        /*
+        Collection names:
 
-    if (name.indexOf('$') !== -1) {
-        return new Error('Contains "$"');
-    }
+        - empty string is not valid
+        - cannot contain "\0"
+        - avoid creating any collections with "system." prefix
+        - user created collections should not contain "$" in the name
+        - database name + collection name < 100 (actual 120)
+        */
 
-    if (name.length + this.settings.partition.length >= 100) {
-        return new Error('Segment and partition name lengths exceeds 100 characters');
-    }
+        if (!name) {
+            return new Error('Empty string');
+        }
 
-    return null;
-};
+        if (name.indexOf('\0') !== -1) {
+            return new Error('Includes null character');
+        }
 
+        if (name.indexOf('system.') === 0) {
+            return new Error('Begins with "system."');
+        }
 
-internals.Connection.prototype.getCollection = function (name, callback) {
+        if (name.indexOf('$') !== -1) {
+            return new Error('Contains "$"');
+        }
 
-    if (!this.isConnected) {
-        return callback(new Error('Connection not ready'));
-    }
+        if (name.length + this.settings.partition.length >= 100) {
+            return new Error('Segment and partition name lengths exceeds 100 characters');
+        }
 
-    if (this.collections[name]) {
-        return callback(null, this.collections[name]);
+        return null;
     }
 
-    // Fetch collection
-
+    async getCollection(name) {
 
-    this.db.collection(name, (err, collection) => {
+        if (!this.isConnected) {
+            throw new Error('Connection not ready');
+        }
 
-        if (err) {
-            return callback(err);
+        if (!name) {
+            throw new Error('Collection name missing');
         }
 
-        // Found
-        collection.ensureIndex({ expiresAt: 1 }, { expireAfterSeconds: 0 }, (err) => {
+        if (this.collections[name]) {
+            return this.collections[name];
+        }
 
-            if (err) {
-                return callback(err);
-            }
+        // Fetch collection
+        const collection = await this.db.collection(name);
 
+        // Found
+        try {
+            await collection.ensureIndex({ expiresAt: 1 }, { expireAfterSeconds: 0 });
             this.collections[name] = collection;
-            return callback(null, collection);
-        });
-    });
-};
 
-
-internals.Connection.prototype.get = function (key, callback) {
-
-    if (!this.isConnectionStarted) {
-        return callback(new Error('Connection not started'));
+            return collection;
+        }
+        catch (err) {
+            throw err;
```

Comment:
I have not used the await yet as much. But does make sense to actually rethrow here like that? If you would remove the try catch here - would it make any difference?

Reply:
Just let it go without catching. No need to catch just to throw the same thing again.